src/sql/database.sql

```diff
@@ -17,22 +17,27 @@ CREATE TABLE `categories` (
   `categoryId` int NOT NULL AUTO_INCREMENT,
   `title` varchar(100) NOT NULL DEFAULT '',
   `description` text,
-  PRIMARY KEY (`categoryId`)
+  `userId` int DEFAULT NULL,
+  PRIMARY KEY (`categoryId`),
+  KEY `fk_users_cats_idx` (`userId`),
+  CONSTRAINT `fk_users_cats` FOREIGN KEY (`userId`) REFERENCES `users` (`userId`)
 ) ENGINE=InnoDB AUTO_INCREMENT=3 DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_0900_ai_ci;
 
 DESCRIBE categories;
 
 DROP TABLE IF EXISTS `tasks`;
+
+CREATE TABLE `tasks` (
   `taskId` int NOT NULL AUTO_INCREMENT,
   `title` varchar(100) NOT NULL DEFAULT '',
   `description` text,
   `categoryId` int DEFAULT NULL,
   `userId` int DEFAULT NULL,
   PRIMARY KEY (`taskId`),
-  KEY `fk_user_id_idx` (`userId`),
-  KEY `fk_category_id_idx` (`categoryId`),
-  CONSTRAINT `fk_category_id` FOREIGN KEY (`categoryId`) REFERENCES `categories` (`categoryId`) ON DELETE RESTRICT ON UPDATE CASCADE,
-  CONSTRAINT `fk_user_id` FOREIGN KEY (`userId`) REFERENCES `users` (`userId`) ON DELETE RESTRICT ON UPDATE CASCADE
+  KEY `fk_users_idx` (`userId`),
+  KEY `fk_categories_idx` (`categoryId`),
```

Comment:
very cool, just don't forget that *task* should be able to to have many *Categories* and not just once.

Reply:
@edmret  any idea of how I can create the connection that allows me to create many categories related to task?

Reply:
@kukuduran05 , you should create another table that have bot foreign keys. looks like this video explains consistently : https://www.youtube.com/watch?v=1eUn6lsZ7c4&ab_channel=DatabaseStar